docs/instrumentations/golang/golang.mdx

```diff
@@ -19,7 +19,10 @@ The following go modules will be auto instrumented by Odigos:
 
 ### HTTP Frameworks
 
-- [`github.com/gin-gonic/gin`](https://pkg.go.dev/github.com/gin-gonic/gin) <sup>*</sup>. HTTP web framework
+<Warning>Please note that in order for the `http.route` parameter to be properly extracted by Go auto instrumentation, 
+you should use Go 1.22 or above.</Warning>
```

Comment:
This section contains 2 list items, but the comment is only relevant to `net/http`. we need to structure it correctly